lms/djangoapps/django_comment_client/forum/tests.py

```diff
@@ -278,6 +291,54 @@ def test_not_found(self, mock_request):
         )
 
 
+@ddt.ddt
+@patch('requests.request')
+@override_settings(MODULESTORE=TEST_DATA_MONGO_MODULESTORE)
+class SingleThreadQueryCountTestCase(ModuleStoreTestCase):
+    """
+    Ensures the number of modulestore queries is deterministic based on the
+    number of responses retrieved for a given discussion thread.
+    """
+
+    @ddt.data(
+        # old mongo: number of responses plus 16.  TODO: O(n)!
+        (ModuleStoreEnum.Type.mongo, 1, 17),
+        (ModuleStoreEnum.Type.mongo, 50, 66),
```

Comment:
Given that this is `O(n)`, why wasn't it broken before? Maybe just because retrieving the structure document is considerably more expensive than just loading up a single course document?